projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -43,13 +44,26 @@ export class AuthService {
    */
   async checkOAuthParamsInUrl(): Promise<void> {
     try {
+      const tokenReceived = getLastValueSync(
+        this.oAuthLibWrapperService.events$.pipe(
+          filter((e) => e.type === 'token_received')
+        )
+      );
```

Comment:
I have a problem to understand why we have used here `getLastValueSync`.
https://github.com/SAP/spartacus/blob/5df4e4c92ae3c24eb4df27b79b51b46f36fb732b/projects/core/src/util/rxjs/get-last-value-sync.ts#L3-L17
As far as I understand, the event `token_reveived` has a chance to occur only later, during the invocation of `tryLogin()`. But, because we attempt to pick the value from the observable synchronously before the `tryLogin() `is called, IMHO the const `tokenReceived` will be always falsy.

--
Instead, did you consider subscribing to the events$ observable (+ `pipe(filter(...), take(1))`) first, and checking if it emitted anything only lately, after we have called `tryLogin()`?

PS. It would be nice to add some typing for `tokenReceived`